langserver/config.go

```diff
@@ -137,41 +154,91 @@ func readConfigFromENV() (*Config, error) {
 
 // DidChangeConfiguration is required by the protocol.Server interface.
 func (s *server) DidChangeConfiguration(ctx context.Context, params *protocol.DidChangeConfigurationParams) error {
-	langserverAddressConfigPath := []string{"promql", "url"}
+	if params == nil {
+		return nil
+	}
+	// nolint: errcheck
+	s.client.LogMessage(
+		s.lifetime,
+		&protocol.LogMessageParams{
+			Type:    protocol.Info,
+			Message: fmt.Sprintf("Received notification change: %v\n", params),
+		})
+
+	setting := params.Settings
+
+	// the struct expected is the following
+	// promql:
+	//   url: http://
+	//   interval: 3w
+	m, ok := setting.(map[string]map[string]string)
+	if !ok {
+		// nolint: errcheck
+		s.client.LogMessage(ctx, &protocol.LogMessageParams{
+			Type:    protocol.Error,
+			Message: fmt.Sprint("unexpected format of the configuration"),
```

Comment:
The format we get here is depends on the language client.

To handle this gracefully we might want to only support `DidChangeConfiguration` with using VS Code for now.

For supporting Sublime, we'd have to know how the `DidChangeConfiguration` request from there can look like. Maybe @nevill can help with this.

Reply:
`LSP-promql` doesn't send `didChangeConfiguration` notification, so it's safe to change the format here.

I see there are [arguments](https://github.com/Microsoft/language-server-protocol/issues/567) on the usage of it.
But in any case, if we support this feature, we'd better document the settings somewhere.

Reply:
yeah it should be documented in this readme at least. And besides it's a bit weird the names of the different value are not the same as the name in the configuration file.

Reply:
besides here for the moment I keep a bit the way to get the data one by one using an hardcoding string for each. But so far, the best way to handle it would be to create another struct and then unmarshall it properly using the json package

Reply:
We should have an extra document explaining how to integrate the language server with new editors.